Framework/API/src/Algorithm.cpp

```diff
@@ -32,6 +32,9 @@
 
 #include <map>
 
+// Index property handling template definitions
+#include "MantidAPI/Algorithm.tcc"
```

Comment:
Unless you put explicit instantiations somewhere you will be forced to include this in all algorithms using the new property.